hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -275,6 +280,11 @@ public QueryResult newQueryResult(int numberOfPartitions) {
         return new QueryResult(queryResultSizeLimiter.getNodeResultLimit(numberOfPartitions));
     }
 
+    @Override
+    public Predicate optimize(Predicate predicate, Indexes indexes) {
```

Comment:
in theory - yes. in practice - no.
you want to minimize number of optimization attempts as it can be quite costly.